pkgs/by-name/gk/gk-cli/package.nix

```
@@ -0,0 +1,50 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  coreutils,
+  installShellFiles,
+  makeWrapper,
+}:
+
+let
+  version = "2.1.0";
+in
+stdenv.mkDerivation {
+  pname = "gk-cli";
+  inherit version;
```

There seems to be a preference for use of `finalAttrs` rather than `let … in` in the packages that I've come across:
```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "gk-cli";
  version = "2.1.0";
```

---

i think it's the first time i see this pattern

what are the differences ? why `finalAttrs` is preferred over `let ... in` ?

---

Unfortunately I don't have a well-educated answer for that, @pinage404. It's one of the many areas where I can deepen knowledge about nixpkgs myself.

In the [documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/using/overrides.chapter.md#pkgoverrideattrs-sec-pkg-overrideattrs) I found the following:

> The argument finalAttrs refers to the final attributes passed to mkDerivation, plus the finalPackage attribute which is equal to the result of mkDerivation or subsequent overrideAttrs calls.